pkg/operator/controller/status/controller.go

```diff
@@ -168,23 +179,41 @@ func initializeClusterOperator(co *configv1.ClusterOperator) {
 	}
 }
 
+type operatorState struct {
+	Namespace          *corev1.Namespace
+	IngressControllers []operatorv1.IngressController
+	DNSRecords         []iov1.DNSRecord
+}
+
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState(nsName string) ([]operatorv1.IngressController, *corev1.Namespace, error) {
+func (r *reconciler) getOperatorState(nsName string) (*operatorState, error) {
+	state := &operatorState{}
+
 	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: nsName}, ns); err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil, nil
+		if !errors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get namespace %q: %v", nsName, err)
 		}
-		return nil, nil, fmt.Errorf("error getting Namespace %s: %v", nsName, err)
+	} else {
+		state.Namespace = ns
 	}
 
 	ingressList := &operatorv1.IngressControllerList{}
 	if err := r.cache.List(context.TODO(), ingressList, client.InNamespace(r.Namespace)); err != nil {
-		return nil, nil, fmt.Errorf("failed to list IngressControllers: %v", err)
+		return nil, fmt.Errorf("failed to list ingresscontrollers in %q: %v", r.Namespace, err)
+	} else {
+		state.IngressControllers = ingressList.Items
+	}
+
+	dnsRecords := &iov1.DNSRecordList{}
+	if err := r.cache.List(context.TODO(), ingressList, client.InNamespace(r.Namespace)); err != nil {
```

Comment:
Bad copy pasta: `ingressList`→`dnsRecords`.

Reply:
Hopefully the tests failed for that... thanks